shinkai-libs/shinkai-http-api/src/api_v2/api_v2_handlers_app_files.rs

```diff
@@ -0,0 +1,452 @@
+}
+
+#[utoipa::path(
+    post,
+    path = "/v2/app_file",
+    responses(
+        (status = 200, description = "Successfully uploaded file", body = Value),
+        (status = 400, description = "Bad request", body = APIError),
+        (status = 500, description = "Internal server error", body = APIError)
+    )
+)]
+pub async fn upload_file_handler(
+    sender: Sender<NodeCommand>,
+    authorization: String,
+    tool_id: String,
+    app_id: String,
+    mut form: FormData,
+) -> Result<impl warp::Reply, warp::Rejection> {
+    let bearer = authorization.strip_prefix("Bearer ").unwrap_or("").to_string();
+
+    let mut file_name = String::new();
+    let mut file_data: Option<Vec<u8>> = Some(vec![]);
+
+    while let Ok(Some(part)) = form.try_next().await {
+        match part.name() {
+            "file_name" => {
+                // Convert the part to bytes then to string
+                let mut bytes = Vec::new();
+                let mut stream = part.stream();
+                while let Ok(Some(chunk)) = stream.try_next().await {
+                    bytes.extend_from_slice(chunk.chunk());
+                }
+                file_name = String::from_utf8_lossy(&bytes).into_owned();
+                file_name = safe_folder_name(&file_name, true);
+            }
+            "file" => {
+                // Read file data
+                let mut bytes = Vec::new();
+                let mut stream = part.stream();
+                while let Ok(Some(chunk)) = stream.try_next().await {
+                    bytes.extend_from_slice(chunk.chunk());
+                }
+                file_data = Some(bytes);
+            }
+            _ => {}
+        }
+    }
+
+    if file_name.is_empty() {
+        return Ok(warp::reply::with_status(
+            warp::reply::json(&APIError {
+                code: 400,
+                error: "Missing file name".to_string(),
+                message: "File name is required".to_string(),
+            }),
+            StatusCode::BAD_REQUEST,
+        ));
+    }
+
+    let (res_sender, res_receiver) = async_channel::bounded(1);
+
+    sender
+        .send(NodeCommand::V2ApiUploadAppFile {
+            bearer,
+            tool_id: safe_folder_name(&tool_id, false),
+            app_id: safe_folder_name(&app_id, false),
+            file_name,
+            file_data: file_data.unwrap(),
+            res: res_sender,
+        })
+        .await
+        .map_err(|_| warp::reject::reject())?;
+
+    let result = res_receiver.recv().await.map_err(|_| warp::reject::reject())?;
+
+    match result {
+        Ok(response) => {
+            let response = create_success_response(response);
+            Ok(warp::reply::with_status(warp::reply::json(&response), StatusCode::OK))
+        }
+        Err(error) => Ok(warp::reply::with_status(
+            warp::reply::json(&error),
+            StatusCode::from_u16(error.code).unwrap(),
+        )),
+    }
+}
+
+// #[derive(Deserialize, ToSchema)]
```

Comment:
delete?

Reply:
Thanks for the catch. I changed from json to multi part and forgot to update the utopia